submodule.go

```diff
@@ -244,7 +244,9 @@ func (s *Submodule) doRecursiveUpdate(ctx context.Context, r *Repository, o *Sub
 func (s *Submodule) fetchAndCheckout(
 	ctx context.Context, r *Repository, o *SubmoduleUpdateOptions, hash plumbing.Hash,
 ) error {
-	if !o.NoFetch {
+	// When SingleBranch is true, skip the broad fetch that retrieves all branches.
+	// Instead, we'll fetch only the specific commit below, reducing network traffic.
+	if !o.NoFetch && !o.SingleBranch {
```

Comment:
This does not seem correct, wouldn't we need to amend the refspec, as per the same [logic in repository](https://github.com/go-git/go-git/blob/71791d86bb6dd549e00f947b339fe0a36f983824/repository.go#L1047-L1068)?